docs/hacking_on_native_stack_unwinding_without_frame_pointers.md

```diff
@@ -0,0 +1,248 @@
+## Hacking on `.eh_frame`-based stack unwinding
+
+Tracking this feature in https://github.com/parca-dev/parca-agent/issues/768.
+
+### Design
+
+The DWARF unwind unformation is read from the `.eh_frame` section (tracking other sections in https://github.com/parca-dev/parca-agent/issues/617), parsed, and evaluated to generate unwind tables (see `table.go`, among others). The unwind tables have, for every program counter in an executable, instructions on how to find the stack pointer value before calling the function of the current frame, as well as information on where to find the return address for the current function, as well as how to calculate the value of various registers in the previous frame.
+
+Once we have these tables in memory, we sort them by program counter, and load them in a BPF map.
+
+Once the BPF program has this table, in order to unwind the stack, it will:
+
+1. Fetches the initial registers
+   1. The instruction pointer `$rip`. Needed to find the row in the unwind table.
+   2. The stack pointer `$rsp`, and the frame pointer `$rbp`, needed to calculate the stack pointer value for the previous frame (CFA). We can find the return address and other registers pushed on the stack at an offset from CFA.
+2. While `unwound_frame_count <= MAX_STACK_DEPTH`
```

Comment:
Typo: unwind_frame_count

Reply:
I think "unwound" is the correct simple past tense for "unwind"

Reply:
Ah, you're going for the past tense. I almost always think in imperative mode. Either way no biggie :)